src/firebase_functions/options.py

```diff
@@ -314,6 +314,42 @@ def _endpoint(
             **_typing.cast(_typing.Dict, kwargs_merged))
 
 
+@_dataclasses.dataclass(frozen=True, kw_only=True)
+class StorageOptions(RuntimeOptions):
+    """
+    Options specific to Storage function types.
+    Internal use only.
+    """
+
+    bucket: _typing.Optional[str] = None
+    """
+    The name of the bucket to watch for Storage events.
+    """
+
+    def _endpoint(
+        self,
+        **kwargs,
+    ) -> _manifest.ManifestEndpoint:
+        assert kwargs["event_type"] is not None
+        # TODO get bucket from admin sdk if no bucket set
+        event_filters: _typing.Any = {
+            "bucket": self.bucket,
+        }
```

Comment:
Let's tackle this as part of this PR? Would love to have storage available by EAP but okay if not available for internal bugbash.